From: Ilya Dryomov <idryomov@gmail.com>
Date: Wed, 2 May 2018 20:16:56 +0200
Subject: iov_iter: fix return type of __pipe_get_pages()
Git-repo: git://git.kernel.org/pub/scm/linux/kernel/git/viro/vfs.git
Git-commit: e76b6312391bdd62e31dc86cb65e478b07b7909e
Patch-mainline: Queued in subsystem maintainer repository
References: bsc#1092710

It returns -EFAULT and happens to be a helper for pipe_get_pages()
whose return type is ssize_t.

Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 lib/iov_iter.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/iov_iter.c
+++ b/lib/iov_iter.c
@@ -970,7 +970,7 @@ unsigned long iov_iter_gap_alignment(con
 }
 EXPORT_SYMBOL(iov_iter_gap_alignment);
 
-static inline size_t __pipe_get_pages(struct iov_iter *i,
+static inline ssize_t __pipe_get_pages(struct iov_iter *i,
 				size_t maxsize,
 				struct page **pages,
 				int idx,
